internal/api/v1/user/transactions/transactions_api_test.go

```diff
@@ -15,6 +15,102 @@ import (
+			draftTransactionResponder:  httpmock.NewJsonResponderOrPanic(http.StatusBadRequest, transactionstest.NewBadRequestSPVError()),
+			recordTransactionResponder: httpmock.NewJsonResponderOrPanic(http.StatusBadRequest, transactionstest.NewBadRequestSPVError()),
+		},
+		"HTTP POST /api/v1/transactions/drafts & /api/v1/transactions response: 500": {
+			expectedErr:                transactionstest.NewInternalServerSPVError(),
+			draftTransactionResponder:  httpmock.NewJsonResponderOrPanic(http.StatusInternalServerError, transactionstest.NewInternalServerSPVError()),
+			recordTransactionResponder: httpmock.NewJsonResponderOrPanic(http.StatusInternalServerError, transactionstest.NewInternalServerSPVError()),
+		},
+	}
+
+	drafTransactionURL := spvwallettest.TestAPIAddr + "/api/v1/transactions/drafts"
+	recordTransactionURL := spvwallettest.TestAPIAddr + "/api/v1/transactions"
+	opReturn := &response.OpReturn{StringParts: []string{"hello", "world"}}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			// given:
+			spvWalletClient, transport := spvwallettest.GivenSPVUserAPI(t)
+			transport.RegisterResponder(http.MethodPost, drafTransactionURL, tc.draftTransactionResponder)
+			transport.RegisterResponder(http.MethodPost, recordTransactionURL, tc.recordTransactionResponder)
+			ctx := context.Background()
+
+			// when:
+			result, err := spvWalletClient.SendToRecipients(ctx, &commands.SendToRecipients{
+				Recipients: []*commands.Recipients{
+					{
+						OpReturn: opReturn,
+					},
+				},
+			})
+
+			// then:
+			require.ErrorIs(t, err, tc.expectedErr)
+			require.Equal(t, tc.expectedResponse, result)
+		})
+	}
+}
+
+func TestTransactionsAPI_FinalizeTransaction(t *testing.T) {
+	tests := map[string]struct {
+		draft       *response.DraftTransaction
+		expectedHex string
+		expectedErr string
+	}{
+		"Finalize Transaction with proper draft": {
+			draft:       transactionstest.ExpectedDraftTransactionWithHex(t),
+			expectedHex: "01000000014c037d55e72d2ee6a95ff67bd758c4cee9c7545bb4d72ba77584152fcfa07012000000006b483045022100a01c25ad9a306f747d90a6d0e815795416ee1f004f865b0653ae3eb2939f42d90220110d994aa99f10533d2566317f55cab838b40f333bf4cdf30c82246461c31fef412102af82c4f5cac25cb5062364937c5e2286094b709610e60b7997b6715784dbf91effffffff0200000000000000000e006a0568656c6c6f05776f726c6408000000000000001976a914702cef80a7039a1aebb70dc05ce1e439646fa33788ac00000000",
+		},
+		"Finalize Transaction fail to parse hex": {
+			draft:       transactionstest.ExpectedDraftTransactionWithWrongHex(t),
+			expectedErr: "failed to parse hex",
+		},
+		"Finalize Transaction fail to prepare locking script": {
+			draft:       transactionstest.ExpectedDraftTransactionWithWrongLockingScript(t),
+			expectedErr: "failed to prepare locking script",
+		},
+		"Finalize Transaction fail to add inputs to transaction": {
+			draft:       transactionstest.ExpectedDraftTransactionWithWrongInputs(t),
+			expectedErr: "failed to add inputs to transaction",
```

Comment:
It’s generally more robust to check errors based on their type rather than their content. Relying on string matching can be fragile — for example, if the error message changes slightly (like adding a comma), the test might fail unnecessarily.